### PR TITLE
Vault 6122 pki role issuer name validation

### PR DESCRIPTION
**Pull request by kitography**
This covers two things:
* @cipherboy 's comments that we should probably say something if the floating reference referred to by a role does not exist. 
* @ivana-mcconnell comments during review that it would be nice if we warned when we were deleting a floating reference to an issuer that was used by a role.

### CODE REVIEW

File: builtin/logical/pki/storage.go [hunk 1]
```diff
@@ -27,6 +27,9 @@ const (
 
 	// Used as a quick sanity check for a reference id lookups...
 	uuidLength = 36
+
+	maxRolesToScanOnIssuerChange = 10000
```
Comment (cipherboy) [inline builtin/logical/pki/storage.go:31]: nit: maybe drop this to 50 or so? This could incur a lot of overhead. /cc @sgmiller

**Review by cipherboy — COMMENTED** [review]
Looks fairly good. It looks like the storage migration test [expects a nil response](https://github.com/hashicorp/vault/blob/main/builtin/logical/pki/storage_migrations_test.go#L175), but we now return a warning. 

Strictly, I think we should perhaps condition the resolution of issuer reference (on roles) to being on the newer storage version? My 2c.

File: builtin/logical/pki/path_roles.go [hunk 2]
```diff
@@ -726,6 +726,18 @@ func (b *backend) pathRoleCreate(ctx context.Context, req *logical.Request, data
 	if len(entry.Issuer) == 0 {
 		entry.Issuer = defaultRef
 	}
+	// Check that the issuers reference set resolves to something
+	issuerId, err := resolveIssuerReference(ctx, req.Storage, entry.Issuer)
```
Comment (stevendpclark) [inline builtin/logical/pki/path_roles.go:730]: Missing err check here.

File: builtin/logical/pki/path_fetch_issuers.go [hunk 3]
```diff
@@ -399,15 +406,22 @@ func (b *backend) pathDeleteIssuer(ctx context.Context, req *logical.Request, da
 		return logical.ErrorResponse("unable to resolve issuer id for reference: " + issuerName), nil
 	}
 
+	var response *logical.Response
+	response = &logical.Response{}
+
+	issuer, err := fetchIssuerById(ctx, req.Storage, ref)
```
Comment (stevendpclark) [inline builtin/logical/pki/path_fetch_issuers.go:412]: Missing err check

File: builtin/logical/pki/storage.go [hunk 4]
```diff
@@ -889,3 +892,37 @@ func isKeyInUse(keyId string, ctx context.Context, s logical.Storage) (inUse boo
...
+	}
+
+	inUseBy = 0
+	checkedRoles := 0
+
+	for _, roleName := range roleEntries {
+		entry, err := storage.Get(ctx, "role/"+roleName)
+		if err != nil {
+			return false, 0, err
+		}
```
Comment (cipherboy) [inline builtin/logical/pki/storage.go:909]: And I think this is the missing `nil` check that semgrep is complaining about.

**Review by stevendpclark — APPROVED** [review]
👍

**Pull request merged**